python/KMS_IMPORTED_KEY_EXPIRATION_CHECK/KMS_IMPORTED_KEY_EXPIRATION_CHECK.py

```diff
@@ -0,0 +1,437 @@
+    And: The number days left before expiration is greater than daysToExpiration value
+   Then: Return COMPLIANT
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::KMS::Key'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    evaluations = []
+    kms_client = get_client('kms', event)
+    keys = list_all_keys(kms_client)
+    for key in keys:
+        key_metadata = kms_client.describe_key(KeyId=key)
+        if is_imported_and_expirable(key_metadata):
+            expiration_date = key_metadata['KeyMetadata']['ValidTo']
+            days_to_expiration = get_days_to_expiration(expiration_date)
+            if days_to_expiration <= valid_rule_parameters['daysToExpiration']:
+                evaluations.append(build_evaluation(key, 'NON_COMPLIANT', event, annotation='The KMS key expires in less than %d days.'%(valid_rule_parameters['daysToExpiration'])))
+            else:
+                evaluations.append(build_evaluation(key, 'COMPLIANT', event))
+    if evaluations:
+        return evaluations
+    return None
+
+def list_all_keys(kms_client):
+    # Returns a list of all KMS key Ids in the region.
+    all_keys = []
+    list_keys_paginator = kms_client.get_paginator('list_keys')
+    list_keys_pages = list_keys_paginator.paginate(Limit=1000)
+    for page in list_keys_pages:
+        all_keys.extend(list(map(lambda key_data: key_data['KeyId'], page['Keys'])))
+    return all_keys
+
+def get_days_to_expiration(expiration_date):
+    # Return the difference between current system datetime object and the expiration datetime object in days.
+    current_date_time = datetime.datetime.now(expiration_date.tzinfo)
+    days = (expiration_date - current_date_time).days
+    return days
+
+def is_imported_and_expirable(key_metadata):
+    # Return True if the key's origin is external and the key material is set to expire.
+    return 'KeyMetadata' in key_metadata and  key_metadata['KeyMetadata']['Origin'] == 'EXTERNAL' and key_metadata['KeyMetadata']['ExpirationModel'] == 'KEY_MATERIAL_EXPIRES'
+
+def evaluate_parameters(rule_parameters):
+    valid_rule_parameters = {}
+    if 'daysToExpiration' not in rule_parameters or not rule_parameters['daysToExpiration']:
+        raise ValueError('daysToExpiration must be provided as a parameter.')
+    days_to_expiration = rule_parameters['daysToExpiration'].lstrip('0')
+    if days_to_expiration.isdigit() and len(days_to_expiration) in range(1, 4) and int(days_to_expiration) in range(366):

```

Comment:
Just check that it is inferior to 999999999. This is the max value supported by time delta function.